packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -163,10 +163,27 @@ export default class ClassTransformer {
     //
     body.push(t.returnStatement(t.cloneNode(this.classRef)));
 
-    const container = t.arrowFunctionExpression(
-      closureParams,
-      t.blockStatement(body),
+    const parent = this.path.findParent(
+      path => path.isProgram() || path.isBlockStatement(),
     );
+    const parentHasUseStrict = parent.node.directives.some(
+      directive => directive.value.value === "use strict",
+    );
+
+    let container;
+
+    if (parentHasUseStrict) {
+      container = t.arrowFunctionExpression(
+        closureParams,
+        t.blockStatement(body),
+      );
+    } else {
+      container = t.arrowFunctionExpression(
+        closureParams,
+        t.blockStatement(body, [t.directive(t.directiveLiteral("use strict"))]),
+      );
+    }
```

Comment:
These two blocks looks really similar, maybe we could add on a condition for the injection of the directive.

Something like:

```js
const directives = [];

if (parentHasUseStrict) {
    directives push t.directive(t.directiveLiteral("use strict"))
}

t.blockStatement(body, directives)
```